modules/core/src/main/java/org/apache/synapse/core/axis2/SynapseCallbackReceiver.java

```diff
@@ -384,7 +385,11 @@ private void handleMessage(String messageID ,MessageContext response,
                     axisOutMsgCtx.getProperty(org.apache.axis2.Constants.OUT_TRANSPORT_INFO));
             response.setTransportIn(axisOutMsgCtx.getTransportIn());
             response.setTransportOut(axisOutMsgCtx.getTransportOut());
-
+            // set "INTERNAL_TRANSACTION_COUNTED" property in response.
+            Object transactionProperty = axisOutMsgCtx.getProperty(BaseConstants.INTERNAL_TRANSACTION_COUNTED);
+            if (transactionProperty instanceof Boolean) {
+                response.setProperty(BaseConstants.INTERNAL_TRANSACTION_COUNTED, transactionProperty);
```

Comment:
I think we can drop the response paths since we are only considering the request in flow?

Reply:
Removing since we are not tracking the response path.